In [1]:
#installing the selenium library
!pip install selenium

In [127]:
#importing all the selenium related library and classes
import selenium
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.common.exceptions import NoSuchElementException,StaleElementReferenceException
import warnings
warnings.filterwarnings('ignore')
import time
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

In [254]:
#Using Chrome Automation
driver=webdriver.Chrome(r'C:\Users\Admin\Downloads\chromedriver_win32 (1)\chromedriver.exe')
driver.get('https://www.amazon.in/') ##taking amazon website

In [245]:
#Extracting reviews,title and ratings for the headsets
srch_items = ['boat rockerz']

title = []

review_text = []

ratings = []


for i in srch_items:

    srchbar = driver.find_element(By.XPATH,"/html/body/div[1]/header/div/div[1]/div[2]/div/form/div[2]/div[1]/input")

    srchbar.send_keys(i)

    driver.find_element(By.XPATH,"/html/body/div[1]/header/div/div[1]/div[2]/div/form/div[3]/div/span/input").click() 

    time.sleep(1)

    item_url = ['https://www.amazon.in/boAt-Rockerz-255-Pro-Earphones/dp/B08TV2P1N8/ref=sr_1_2?crid=2F4TJRPLYFS4O&keywords=boat+rockerz+255+pro+plus&qid=1669960831&qu=eyJxc2MiOiIzLjI3IiwicXNhIjoiMi44NSIsInFzcCI6IjIuNzQifQ%3D%3D&sprefix=boat+rockerz+%2Caps%2C1375&sr=8-2']

    for i in driver.find_elements("xpath","//h2[@class='a-size-mini a-spacing-none a-color-base s-line-clamp-2']/a"):

        item_url.append(i.get_attribute('href'))

    for i in item_url:

        driver.get(i)

        time.sleep(1)

        for _ in range(2):

            driver.execute_script("window.scrollBy(0,1000)")

            time.sleep(1)

        # Click to see all the reviews

        try:

            driver.find_element("xpath","//a[@class='a-link-emphasis a-text-bold']").click()

        except:

            pass

        

        url_ = []

        try:

            two_str = driver.find_element("xpath","//table[@id='histogramTable']/tbody/tr[4]/td[2]/a")

            url_.append(two_str.get_attribute('href'))

        except: pass

        try:

            three_str = driver.find_element("xpath","//table[@id='histogramTable']/tbody/tr[3]/td[2]/a")

            url_.append(three_str.get_attribute('href'))

        except: pass

        try:

            one_str = driver.find_element("xpath","//table[@id='histogramTable']/tbody/tr[5]/td[2]/a")

            url_.append(one_str.get_attribute('href'))

        except: pass

        try:

            fiv_str = driver.find_element("xpath","//table[@id='histogramTable']/tbody/tr[1]/td[2]/a")

            url_.append(fiv_str.get_attribute('href'))

        except: pass

        try:

            four_str = driver.find_element("xpath","//table[@id='histogramTable']/tbody/tr[2]/td[2]/a")

            url_.append(four_str.get_attribute('href'))

        except: pass

        for i in url_:

            driver.get(i)

            time.sleep(1)

            for i in range(150):

                ids = driver.find_elements("xpath","//div[@class='a-section a-spacing-none review-views celwidget']/div")

                comment_ids = []

                for i in ids:

                    comment_ids.append(i.get_attribute('id'))



                for x in comment_ids:

                    # Extracting user ids from each user on a page

                    try:

                        review_title = driver.find_element("xpath",'//*[@id="' + x +'"]/div/div/div[2]/a[2]/span[1]')

                        title.append(review_title.get_attribute("innerHTML"))

                    except NoSuchElementException:

                        title.append('')



                    try:

                        rating = driver.find_element("xpath",'//*[@id="' + x +'"]/div/div/div[2]/a[1]/i/span[1]')

                        ratings.append(rating.get_attribute("innerHTML").split(' ')[0])

                    except NoSuchElementException:

                        ratings.append('')



                    try:

                        text = driver.find_element("xpath",'//*[@id="' + x +'"]/div/div/div[4]/span/span')

                        review_text.append(text.get_attribute("innerHTML"))

                    except NoSuchElementException:

                        review_text.append('')

                try:

                    driver.find_element("xpath","//ul[@class='a-pagination']/li[2]/a").click()

                    time.sleep(3)

                except : break

StaleElementReferenceException: Message: stale element reference: element is not attached to the page document
  (Session info: chrome=107.0.5304.108)
Stacktrace:
Backtrace:
	Ordinal0 [0x007D1ED3+2236115]
	Ordinal0 [0x007692F1+1807089]
	Ordinal0 [0x006766FD+812797]
	Ordinal0 [0x006792B4+823988]
	Ordinal0 [0x00679165+823653]
	Ordinal0 [0x00679AF5+826101]
	Ordinal0 [0x006D6452+1205330]
	Ordinal0 [0x006C1A8C+1120908]
	Ordinal0 [0x006D59E2+1202658]
	Ordinal0 [0x006C18A6+1120422]
	Ordinal0 [0x0069A73D+960317]
	Ordinal0 [0x0069B71F+964383]
	GetHandleVerifier [0x00A7E7E2+2743074]
	GetHandleVerifier [0x00A708D4+2685972]
	GetHandleVerifier [0x00862BAA+532202]
	GetHandleVerifier [0x00861990+527568]
	Ordinal0 [0x0077080C+1837068]
	Ordinal0 [0x00774CD8+1854680]
	Ordinal0 [0x00774DC5+1854917]
	Ordinal0 [0x0077ED64+1895780]
	BaseThreadInitThunk [0x761DFEF9+25]
	RtlGetAppContainerNamedObjectPath [0x77317BBE+286]
	RtlGetAppContainerNamedObjectPath [0x77317B8E+238]


In [246]:
title

['',
 'Very bad experience',
 'मेरा product रिटर्न ले गया लेकिन रसीद नहीं दिया',
 'Not up to the mark, disappointing',
 'Product quality, Sound quality, comfortness',
 'Never buy wireless from boat',
 'Good for 1 year because the product I purchased it do not show actual battery percentage .',
 'DOA product',
 'Sound is not clear at all',
 'Worst product worst company',
 'Silicon ear hooks loose and falls off',
 '',
 '',
 "Nahhh don't buy",
 'Disappointed',
 'Need to improve sound quality. Earbuds are of worst quality.',
 'It was awesome for one month then mic stopped working disappointed.',
 'Miscellaneous trouble',
 'Look like software issue',
 'बेकार प्रोडक्ट है एक साल में ही खराब हो गया',
 'but sound quality achi h.800/- tak thik value h',
 'No charger cable available with the product',
 'Problem with button',
 '',
 '',
 'Bad experience',
 'Not bad',
 'Not fitted',
 'Average product',
 "Cons are much higher than Pros...Don't Go For That...Try other Neckband....",
 'Not so good.',
 

In [247]:
len(title)

11121

In [248]:
review_text

['',
 "Anyone looking to buy this product please don't buy it . The connectivity is very poor even after connecting to the device it continuously ask to pair which is very irritating.when you open the Bluetooth a continuous notification goes to the phone other than the Phone to which it is connected which becomes very frustrating.",
 'मेरा पार्सल डिलिवरी बॉय २४/११/२२.को ले गया लेकिन, रसीद नहीं दिया,२५/११को दूसरा डिलिवरी बॉय पार्सल लेकर आया था और वह रिटर्न पार्सल मांग रहा था. मैं कहासे दु, अब आप बताइए मैं क्या करू,जो पहला बॉय पार्सल ले गया उसका c.c.tv फुटेज मिल सकता है आपको. मुझे मेरा product कब मिलेगा,,या मैं पुलिस में जाऊ आप बताइए.<br>.मेरे कैश रिटर्न के लिए Ac, no Canara Bank 86621180000542',
 "Everything is good but this product not up to the mark, as it is disconnecting by default when not in use, suddenly trying to take a call, you will find that, it is already disconnected, which is very disappointing. I will never by it again, it's seems like waste of money.",
 "Product quality 

In [249]:
len(review_text)

11121

In [250]:
ratings

['',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0

In [251]:
len(ratings)

11121

In [255]:
#Extracting reviews,title and ratings for the routers
srch_items = ['Tp link N300 router']

title1 = []

review_text1 = []

ratings1 = []


for i in srch_items:

    srchbar = driver.find_element(By.XPATH,"/html/body/div[1]/header/div/div[1]/div[2]/div/form/div[2]/div[1]/input")

    srchbar.send_keys(i)

    driver.find_element(By.XPATH,"/html/body/div[1]/header/div/div[1]/div[2]/div/form/div[3]/div/span/input").click() 

    time.sleep(1)

    item_url = ['https://www.amazon.in/TP-Link-TL-WR845N-300Mbps-Wireless-N-Router/dp/B01HGCLUH6/ref=sr_1_1_sspa?keywords=Tp+link+N300+router&qid=1669984256&qu=eyJxc2MiOiIxLjczIiwicXNhIjoiMS4wMyIsInFzcCI6IjAuNDIifQ%3D%3D&sr=8-1-spons&sp_csd=d2lkZ2V0TmFtZT1zcF9hdGY&psc=1']

    for i in driver.find_elements("xpath","//h2[@class='a-size-mini a-spacing-none a-color-base s-line-clamp-2']/a"):

        item_url.append(i.get_attribute('href'))

    for i in item_url:

        driver.get(i)

        time.sleep(1)

        for _ in range(2):

            driver.execute_script("window.scrollBy(0,1000)")

            time.sleep(1)

        # Click to see all the reviews

        try:

            driver.find_element("xpath","//a[@class='a-link-emphasis a-text-bold']").click()

        except:

            pass

        

        url_ = []

        try:

            two_str = driver.find_element("xpath","//table[@id='histogramTable']/tbody/tr[4]/td[2]/a")

            url_.append(two_str.get_attribute('href'))

        except: pass

        try:

            three_str = driver.find_element("xpath","//table[@id='histogramTable']/tbody/tr[3]/td[2]/a")

            url_.append(three_str.get_attribute('href'))

        except: pass

        try:

            one_str = driver.find_element("xpath","//table[@id='histogramTable']/tbody/tr[5]/td[2]/a")

            url_.append(one_str.get_attribute('href'))

        except: pass

        try:

            fiv_str = driver.find_element("xpath","//table[@id='histogramTable']/tbody/tr[1]/td[2]/a")

            url_.append(fiv_str.get_attribute('href'))

        except: pass

        try:

            four_str = driver.find_element("xpath","//table[@id='histogramTable']/tbody/tr[2]/td[2]/a")

            url_.append(four_str.get_attribute('href'))

        except: pass

        for i in url_:

            driver.get(i)

            time.sleep(1)

            for i in range(10):

                ids = driver.find_elements("xpath","//div[@class='a-section a-spacing-none review-views celwidget']/div")

                comment_ids = []

                for i in ids:

                    comment_ids.append(i.get_attribute('id'))



                for x in comment_ids:

                    # Extracting user ids from each user on a page

                    try:

                        review_title = driver.find_element("xpath",'//*[@id="' + x +'"]/div/div/div[2]/a[2]/span[1]')

                        title1.append(review_title.get_attribute("innerHTML"))

                    except NoSuchElementException:

                        title1.append('')



                    try:

                        rating = driver.find_element("xpath",'//*[@id="' + x +'"]/div/div/div[2]/a[1]/i/span[1]')

                        ratings1.append(rating.get_attribute("innerHTML").split(' ')[0])

                    except NoSuchElementException:

                        ratings1.append('')



                    try:

                        text = driver.find_element("xpath",'//*[@id="' + x +'"]/div/div/div[4]/span/span')

                        review_text1.append(text.get_attribute("innerHTML"))

                    except NoSuchElementException:

                        review_text1.append('')

                try:

                    driver.find_element("xpath","//ul[@class='a-pagination']/li[2]/a").click()

                    time.sleep(3)

                except : break

In [256]:
title1

['Better to avoid',
 'Good to use for a small cabin but not for home',
 'Easy installation but Network drop outs there',
 'It is unable to cross wall and it does not have wall mount',
 'Would recommend a different model. This one is ok for a small room only.',
 'started giving issues',
 'Not good',
 'Wifi Signal drops frequently',
 'Performance is poor in 2BHK, go for better option',
 'Wi-Fi signal is not good … signal drops more than 5 to 6 Times in a day',
 '',
 '',
 'Power adaptor not working, one antenna is not flexible not rotating other two are ok',
 'Not a dual band router.',
 'Frequently disconnecting',
 'Max. speed 50 Mbps. Not a good experience.',
 'Easy to setup but not working properly in extender mode.',
 'Abnormal voltage spike due to lighting damages the product.',
 'Never ever think about it.',
 'Range did not increase',
 'This is not Dual Band',
 "Wi-Fi automatically disconnected from every Device's at least 5-6 times every day.",
 '',
 '',
 'DOES NOT HAVE ENOUGH RANGE

In [257]:
len(title1)

6549

In [258]:
review_text1

['I initially purchased this deviced. I had to return this router. But Amazon is awesome. Then I shifted to C80 Archer AC1900 which is a better router in terms of everything like Range, Interface and certainly Firmware. This is a N router and Single band. Was a good one 2 years ago. Not now.,.',
 "Easy to setup.<br>Although my place is only 500 sq.ft, my living room won't receive signal if I keep the router in bedroom. Signal gets dropped multiple times a day. Thinking to contact service desk regarding this issue.",
 'Easy installation. Some time network is dropping. checked with Internet service provider but there is no issue from their side. Seems to be issue with modem, tried to reach the Tech support team but nothing much helped.',
 'It is unable to cross wall and it does not have wall mount',
 'Signal keeps dropping',
 'this router worked for 4 days and started giving issues',
 'My room is 350 sq.ft but the signal is very poor . It’s not reaching every corner. I just put this one 

In [259]:
len(review_text1)

6549

In [260]:
ratings1

['2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '3.0',
 '3.0',
 '3.0',
 '3.0',
 '3.0',
 '3.0',
 '3.0',
 '3.0',
 '3.0',
 '3.0',
 '',
 '',
 '3.0',
 '3.0',

In [261]:
len(ratings1)

6549

In [263]:
#Extracting reviews,title and ratings for the mobiles
srch_items = ['One Plus Nord']

title2 = []

review_text2 = []

ratings2 = []


for i in srch_items:

    srchbar = driver.find_element(By.XPATH,"/html/body/div[1]/header/div/div[1]/div[2]/div/form/div[2]/div[1]/input")

    srchbar.send_keys(i)

    driver.find_element(By.XPATH,"/html/body/div[1]/header/div/div[1]/div[2]/div/form/div[3]/div/span/input").click()  

    time.sleep(1)

    item_url = ['https://www.amazon.in/OnePlus-Nord-Lite-128GB-Storage/dp/B09WQYFLRX/ref=sr_1_2?crid=S4Y1W5CIS1X9&keywords=oneplus+nord+ce+2+5g&qid=1669993311&qu=eyJxc2MiOiIzLjI5IiwicXNhIjoiMi43NyIsInFzcCI6IjIuNDAifQ%3D%3D&sprefix=one+plus%2Caps%2C407&sr=8-2']

    for i in driver.find_elements("xpath","//h2[@class='a-size-mini a-spacing-none a-color-base s-line-clamp-2']/a"):

        item_url.append(i.get_attribute('href'))

    for i in item_url:

        driver.get(i)

        time.sleep(1)

        for _ in range(2):

            driver.execute_script("window.scrollBy(0,1000)")

            time.sleep(1)

        # Click to see all the reviews

        try:

            driver.find_element("xpath","//a[@class='a-link-emphasis a-text-bold']").click()

        except:

            pass

        

        url_ = []

        try:

            two_str = driver.find_element("xpath","//table[@id='histogramTable']/tbody/tr[4]/td[2]/a")

            url_.append(two_str.get_attribute('href'))

        except: pass

        try:

            three_str = driver.find_element("xpath","//table[@id='histogramTable']/tbody/tr[3]/td[2]/a")

            url_.append(three_str.get_attribute('href'))

        except: pass

        try:

            one_str = driver.find_element("xpath","//table[@id='histogramTable']/tbody/tr[5]/td[2]/a")

            url_.append(one_str.get_attribute('href'))

        except: pass

        try:

            fiv_str = driver.find_element("xpath","//table[@id='histogramTable']/tbody/tr[1]/td[2]/a")

            url_.append(fiv_str.get_attribute('href'))

        except: pass

        try:

            four_str = driver.find_element("xpath","//table[@id='histogramTable']/tbody/tr[2]/td[2]/a")

            url_.append(four_str.get_attribute('href'))

        except: pass

        for i in url_:

            driver.get(i)

            time.sleep(1)

            for i in range(20):

                ids = driver.find_elements("xpath","//div[@class='a-section a-spacing-none review-views celwidget']/div")

                comment_ids = []

                for i in ids:

                    comment_ids.append(i.get_attribute('id'))



                for x in comment_ids:

                    # Extracting user ids from each user on a page

                    try:

                        review_title = driver.find_element("xpath",'//*[@id="' + x +'"]/div/div/div[2]/a[2]/span[1]')

                        title2.append(review_title.get_attribute("innerHTML"))

                    except NoSuchElementException:

                        title2.append('')



                    try:

                        rating = driver.find_element("xpath",'//*[@id="' + x +'"]/div/div/div[2]/a[1]/i/span[1]')

                        ratings2.append(rating.get_attribute("innerHTML").split(' ')[0])

                    except NoSuchElementException:

                        ratings2.append('')



                    try:

                        text = driver.find_element("xpath",'//*[@id="' + x +'"]/div/div/div[4]/span/span')

                        review_text2.append(text.get_attribute("innerHTML"))

                    except NoSuchElementException:

                        review_text2.append('')

                try:

                    driver.find_element("xpath","//ul[@class='a-pagination']/li[2]/a").click()

                    time.sleep(3)

                except : break

In [264]:
title2

['',
 "Android Auto Doesn't Work || UPI Apps don't work on Roaming",
 'Terrible Camera with average battery life!',
 "Worst phone don't buy",
 'Mobile keeps hanging post new update',
 'I want to return this product but the return window has been closed by that time.',
 'Camera/Build quality/Sound Quality',
 'This phone has battery draining issue. Not worth full at  all.',
 'Not value for money',
 'Phone problems.',
 'Bad camera',
 '',
 '',
 'Camera quality is not clear.',
 'Replace phone',
 "Don't purchase this mobile",
 'Bad product. Has heating issues',
 'Mt lena',
 'Very bad battery backup',
 'Over all an average phone',
 'Camera quality poor',
 'Poor user experience and a camera',
 'Camera quality poor',
 '',
 '',
 'Not as promised',
 'असमाधानकारक',
 'My phone battery life.',
 'Good product lovely phone',
 'Do not buy this.',
 'Bekaar Mobile',
 "It's not good as compared to Samsung same range mobile",
 'Not worthy !',
 'Worst Oneplus',
 'Adapter kharab ho gya mera replacement k liy

In [265]:
len(title2)

9251

In [266]:
review_text2

['',
 'Bought it 19 August. Regretting since 20 August.<br><br>Took me a while to figure out how to start Android Auto by default when plugged into a car (you have to manually enable Android Auto the first time). First time I plugged it in I was in a busy road and was unsure which way to go. The phone just wouldn\'t pick up Android Auto. I had to manually configure it later on after watching a couple of videos on YT. Since then, whenever I plug it in my car, I keep getting the pop-up that asks me to choose whether I want to "Use USB to - Charge Only - or Transfer Files/Android Auto - or Transfer Photos". No matter how many times I select Android Auto, the same thing keeps popping up, and Android Auto DOES NOT WORK. Never faced this problem with my old Redmi Note 5 Pro.<br><br>When I was roaming outside my home circle (Kolkata) for a few weeks, the following apps didn\'t work: Zomato, PhonePe, Paytm. I had read before ordering the phone that they don\'t always optimize apps for non-Snap

In [267]:
len(review_text2)

9251

In [268]:
ratings2

['',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0

In [269]:
len(ratings2)

9251

In [223]:
driver=webdriver.Chrome(r'C:\Users\Admin\Downloads\chromedriver_win32 (1)\chromedriver.exe')
driver.get('https://www.amazon.in/')

In [216]:
#Extracting reviews,title and ratings for the printers
srch_items = ['Printers']

title3 = []

review_text3 = []

ratings3 = []

for i in srch_items:

    srchbar = driver.find_element(By.XPATH,"/html/body/div[1]/header/div/div[1]/div[2]/div/form/div[2]/div[1]/input")

    srchbar.send_keys(i)

    driver.find_element(By.XPATH,"/html/body/div[1]/header/div/div[1]/div[2]/div/form/div[3]/div/span/input").click()  

    time.sleep(1)

    item_url = ['https://www.amazon.in/HP-DeskJet-Inkjet-Colour-Printer/dp/B08D9NDZ1Y/ref=sr_1_3?crid=383GWF1ZZWV3M&keywords=printers&qid=1669917667&qu=eyJxc2MiOiI1Ljg1IiwicXNhIjoiNS41NSIsInFzcCI6IjQuNjYifQ%3D%3D&sprefix=printer%2Caps%2C375&sr=8-3'] # using url of the product directly



    for i in item_url:

        driver.get(i)

        time.sleep(1)

        for _ in range(2):

            driver.execute_script("window.scrollBy(0,1000)")

            time.sleep(1)

        # Click to see all the reviews

        try:

            driver.find_element("xpath","//a[@class='a-link-emphasis a-text-bold']").click()

        except:

            pass

        

        url_ = []

        try:

            two_str = driver.find_element("xpath","//table[@id='histogramTable']/tbody/tr[4]/td[2]/a")

            url_.append(two_str.get_attribute('href'))

        except: pass

        try:

            three_str = driver.find_element("xpath","//table[@id='histogramTable']/tbody/tr[3]/td[2]/a")

            url_.append(three_str.get_attribute('href'))

        except: pass

        try:

            one_str = driver.find_element("xpath","//table[@id='histogramTable']/tbody/tr[5]/td[2]/a")

            url_.append(one_str.get_attribute('href'))

        except: pass

        try:

            fiv_str = driver.find_element("xpath","//table[@id='histogramTable']/tbody/tr[1]/td[2]/a")

            url_.append(fiv_str.get_attribute('href'))

        except: pass

        try:

            four_str = driver.find_element("xpath","//table[@id='histogramTable']/tbody/tr[2]/td[2]/a")

            url_.append(four_str.get_attribute('href'))

        except: pass

        for i in url_:

            driver.get(i)

            time.sleep(1)

            for i in range(120):

                ids = driver.find_elements("xpath","//div[@class='a-section a-spacing-none review-views celwidget']/div")

                comment_ids = []

                for i in ids:

                    comment_ids.append(i.get_attribute('id'))



                for x in comment_ids:

                    # Extracting user ids from each user on a page

                    try:

                        review_title = driver.find_element("xpath",'//*[@id="' + x +'"]/div/div/div[2]/a[2]/span[1]')

                        title3.append(review_title.get_attribute("innerHTML"))

                    except NoSuchElementException:

                        title3.append('')



                    try:

                        rating = driver.find_element("xpath",'//*[@id="' + x +'"]/div/div/div[2]/a[1]/i/span[1]')

                        ratings3.append(rating.get_attribute("innerHTML").split(' ')[0])

                    except NoSuchElementException:

                        ratings3.append('')



                    try:

                        text = driver.find_element("xpath",'//*[@id="' + x +'"]/div/div/div[4]/span/span')

                        review_text3.append(text.get_attribute("innerHTML"))

                    except NoSuchElementException:

                        review_text3.append('')

                try:

                    driver.find_element("xpath","//ul[@class='a-pagination']/li[2]/a").click()

                    time.sleep(3)

                except : break

In [217]:
title3

['HP SMART APP? HP DULL APP! HP TECH Support (INDIA) is of no help! Reaching God would been easier!',
 'Printer cost is low but catrege cost is very high',
 'Cartridge is the one to drill hole in pocket',
 'Constant leakage of cartridge.😠',
 'I made a mistake buying it',
 'Only looks and brand, Not worth it',
 'Ink guzzler, be ready to buy cartridges every month',
 'VERY COSTLY',
 'It gets very high costly after long time',
 'Think before buy',
 '',
 '',
 'After 30 prints..blank print came out.  HP printer  buy date is 9-6-22..',
 'Not good',
 'Very bad product',
 'Very low qty to print.',
 'Not satisfied',
 'There is an issue with the product after just one print th',
 'Printer software and quality much  more better',
 '👎👎👎😡',
 'Very poor this printer lost money',
 'Not so good!',
 '',
 '',
 'Standard A4 size Boarder not working properly',
 'Frequency of changing cartridge',
 'Quality of functions good but poor Durability',
 'Buggy software, slow printing and cartridges depleted too s

In [218]:
len(title3)

5685

In [219]:
review_text3

["I ordered this machine to copy, print, and scan to save images on my PC. I wanted to help my 89-year mother during my three-week stay in India. Copy&amp; print? no problem! Scanning the document and saving the to PC using HP smart? It was a nightmare!  I could not get a single HP tech support person on the phone/Whatsapp /online chat to help. The directions on the brochure were ridiculous. I needed to turn in important financial documents before the deadline and I could not.  I called USA HP support and begged them to help me figure it out. Though the agent was sympathetic, he refused to intervene claiming it was 'HP India's responsibility. I was in tears. I couldn't help my poor mother who is living in a small town and c was counting on me to sort things out for her.<br> I just don't understand why to sell a product that you can't support esp. for a global conglomerate such as HP. Just disappointed!",
 'Printer cost is low but cartridge cost very high 60 page print out only Cartridg

In [220]:
len(review_text3)

5685

In [221]:
ratings3

['2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '

In [222]:
len(ratings3)

5685

In [224]:
#Extracting reviews,title and ratings for the smartwatch
srch_items = ['FitBit Smartwatch']

title4 = []

review_text4 = []

ratings4 = []


for i in srch_items:

    srchbar = driver.find_element(By.XPATH,"/html/body/div[1]/header/div/div[1]/div[2]/div/form/div[2]/div[1]/input")

    srchbar.send_keys(i)

    driver.find_element(By.XPATH,"/html/body/div[1]/header/div/div[1]/div[2]/div/form/div[3]/div/span/input").click()

    time.sleep(1)

    item_url = ['https://www.amazon.in/product-reviews/B07TWFVDWT/ref=acr_dp_hist_2?ie=UTF8&filterByStar=two_star&reviewerType=all_reviews#reviews-filter-bar']

    for i in driver.find_elements("xpath","//h2[@class='a-size-mini a-spacing-none a-color-base s-line-clamp-2']/a"):

        item_url.append(i.get_attribute('href'))

    for i in item_url:

        driver.get(i)

        time.sleep(1)

        for _ in range(2):

            driver.execute_script("window.scrollBy(0,1000)")

            time.sleep(1)

        # Click to see all the reviews

        try:

            driver.find_element("xpath","//a[@class='a-link-emphasis a-text-bold']").click()

        except:

            pass

        

        url_ = []

        try:

            two_str = driver.find_element("xpath","//table[@id='histogramTable']/tbody/tr[4]/td[2]/a")

            url_.append(two_str.get_attribute('href'))

        except: pass

        try:

            three_str = driver.find_element("xpath","//table[@id='histogramTable']/tbody/tr[3]/td[2]/a")

            url_.append(three_str.get_attribute('href'))

        except: pass

        try:

            one_str = driver.find_element("xpath","//table[@id='histogramTable']/tbody/tr[5]/td[2]/a")

            url_.append(one_str.get_attribute('href'))

        except: pass

        try:

            fiv_str = driver.find_element("xpath","//table[@id='histogramTable']/tbody/tr[1]/td[2]/a")

            url_.append(fiv_str.get_attribute('href'))

        except: pass

        try:

            four_str = driver.find_element("xpath","//table[@id='histogramTable']/tbody/tr[2]/td[2]/a")

            url_.append(four_str.get_attribute('href'))

        except: pass

        for i in url_:

            driver.get(i)

            time.sleep(1)

            for i in range(140):

                ids = driver.find_elements("xpath","//div[@class='a-section a-spacing-none review-views celwidget']/div")

                comment_ids = []

                for i in ids:

                    comment_ids.append(i.get_attribute('id'))



                for x in comment_ids:

                    # Extracting user ids from each user on a page

                    try:

                        review_title = driver.find_element("xpath",'//*[@id="' + x +'"]/div/div/div[2]/a[2]/span[1]')

                        title4.append(review_title.get_attribute("innerHTML"))

                    except NoSuchElementException:

                        title4.append('')



                    try:

                        rating = driver.find_element("xpath",'//*[@id="' + x +'"]/div/div/div[2]/a[1]/i/span[1]')

                        ratings4.append(rating.get_attribute("innerHTML").split(' ')[0])

                    except NoSuchElementException:

                        ratings4.append('')



                    try:

                        text = driver.find_element("xpath",'//*[@id="' + x +'"]/div/div/div[4]/span/span')

                        review_text4.append(text.get_attribute("innerHTML"))

                    except NoSuchElementException:

                        review_text4.append('')

                try:

                    driver.find_element("xpath","//ul[@class='a-pagination']/li[2]/a").click()

                    time.sleep(3)

                except : break

StaleElementReferenceException: Message: stale element reference: element is not attached to the page document
  (Session info: chrome=107.0.5304.108)
Stacktrace:
Backtrace:
	Ordinal0 [0x007D1ED3+2236115]
	Ordinal0 [0x007692F1+1807089]
	Ordinal0 [0x006766FD+812797]
	Ordinal0 [0x006792B4+823988]
	Ordinal0 [0x00679165+823653]
	Ordinal0 [0x00679AF5+826101]
	Ordinal0 [0x006D6452+1205330]
	Ordinal0 [0x006C1A8C+1120908]
	Ordinal0 [0x006D59E2+1202658]
	Ordinal0 [0x006C18A6+1120422]
	Ordinal0 [0x0069A73D+960317]
	Ordinal0 [0x0069B71F+964383]
	GetHandleVerifier [0x00A7E7E2+2743074]
	GetHandleVerifier [0x00A708D4+2685972]
	GetHandleVerifier [0x00862BAA+532202]
	GetHandleVerifier [0x00861990+527568]
	Ordinal0 [0x0077080C+1837068]
	Ordinal0 [0x00774CD8+1854680]
	Ordinal0 [0x00774DC5+1854917]
	Ordinal0 [0x0077ED64+1895780]
	BaseThreadInitThunk [0x761DFEF9+25]
	RtlGetAppContainerNamedObjectPath [0x77317BBE+286]
	RtlGetAppContainerNamedObjectPath [0x77317B8E+238]


In [225]:
title4

['Not worth to invest money on ...No service available in India',
 'Worth avoiding',
 'Short product life; Not value for money',
 'Product does not include any manual',
 'Good but Not Reliable!',
 "Not a good quality at all. Within a year, it has stopped working. There's no service too",
 'Device quality',
 'Used one',
 'After 2 years of using it, Its not worth it',
 'bad hardware',
 '',
 '',
 'A very unnecassarily expensive and inaccurate piece of accessory!',
 "Don't go for this not worth",
 'Not worth the money',
 'Not for people who want to track their exercise accurately.',
 'Not worth it',
 "Worth or not worth based on individuals. I wouldn't say worth.",
 'Dissapointed',
 'Not so good and user friendly',
 'worst smart gadget that i have ever purchased from amazon.',
 'This particular Fitbit verse never worked',
 '',
 '',
 'does not sync with Fitbit app on Android',
 'Worst product. Not value for money product compare to other smart watches',
 'Good watch but the connectivity suc

In [226]:
len(title4)

28695

In [229]:
review_text4

['Not worth to put so much money if they cannot provide service.<br>Its not really worth it. If youll get any issue after a year of purchasing , there is no solution available .the number provided on website is not working and not much help will be offered form Customer care -Chat. Only option left is go for aftermarket repair  which is unreliable .',
 "Had purchased last year. Failed exactly after an year. They don't have any service centers in India if there are issues. You end up throwing it.<br>Better avoided.",
 'Fitbit brand seems to be deteriorating; A product which was the innovator and market leader in smart watch industry, now seems to be contend with charging premium pricing basis past glory, but there is a complete lack of attention to product quality and after warranty support. The competition has moved ahead and now comparable products are available at much more attractive pricing.  Hopefully, Fitbit will wake up and do something. The app sync experience is also bad and t

In [230]:
len(review_text4)

28695

In [231]:
ratings4

['2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',
 '',


In [232]:
len(ratings4)

28695

In [270]:
#Extracting reviews,title and ratings for the Bluetooth Devices
srch_items = ['Echo Dot 3rd Generation']

title5 = []

review_text5 = []

ratings5 = []


for i in srch_items:

    srchbar = driver.find_element(By.XPATH,"/html/body/div[1]/header/div/div[1]/div[2]/div/form/div[2]/div[1]/input")

    srchbar.send_keys(i)

    driver.find_element(By.XPATH,"/html/body/div[1]/header/div/div[1]/div[2]/div/form/div[3]/div/span/input").click()  

    time.sleep(1)

    item_url = ['https://www.amazon.in/Echo-Dot-3rd-Gen/dp/B07PFFMP9P/ref=lp_1389339031_1_1']

    for i in driver.find_elements("xpath","//h2[@class='a-size-mini a-spacing-none a-color-base s-line-clamp-2']/a"):

        item_url.append(i.get_attribute('href'))

    for i in item_url:

        driver.get(i)

        time.sleep(1)

        for _ in range(2):

            driver.execute_script("window.scrollBy(0,1000)")

            time.sleep(1)

        # Click to see all the reviews

        try:

            driver.find_element("xpath","//a[@class='a-link-emphasis a-text-bold']").click()

        except:

            pass

        

        url_ = []

        try:

            two_str = driver.find_element("xpath","//table[@id='histogramTable']/tbody/tr[4]/td[2]/a")

            url_.append(two_str.get_attribute('href'))

        except: pass

        try:

            three_str = driver.find_element("xpath","//table[@id='histogramTable']/tbody/tr[3]/td[2]/a")

            url_.append(three_str.get_attribute('href'))

        except: pass

        try:

            one_str = driver.find_element("xpath","//table[@id='histogramTable']/tbody/tr[5]/td[2]/a")

            url_.append(one_str.get_attribute('href'))

        except: pass

        try:

            fiv_str = driver.find_element("xpath","//table[@id='histogramTable']/tbody/tr[1]/td[2]/a")

            url_.append(fiv_str.get_attribute('href'))

        except: pass

        try:

            four_str = driver.find_element("xpath","//table[@id='histogramTable']/tbody/tr[2]/td[2]/a")

            url_.append(four_str.get_attribute('href'))

        except: pass

        for i in url_:

            driver.get(i)

            time.sleep(1)

            for i in range(100):

                ids = driver.find_elements("xpath","//div[@class='a-section a-spacing-none review-views celwidget']/div")

                comment_ids = []

                for i in ids:

                    comment_ids.append(i.get_attribute('id'))



                for x in comment_ids:

                    # Extracting user ids from each user on a page

                    try:

                        review_title = driver.find_element("xpath",'//*[@id="' + x +'"]/div/div/div[2]/a[2]/span[1]')

                        title5.append(review_title.get_attribute("innerHTML"))

                    except NoSuchElementException:

                        title5.append('')



                    try:

                        rating = driver.find_element("xpath",'//*[@id="' + x +'"]/div/div/div[2]/a[1]/i/span[1]')

                        ratings5.append(rating.get_attribute("innerHTML").split(' ')[0])

                    except NoSuchElementException:

                        ratings5.append('')



                    try:

                        text = driver.find_element("xpath",'//*[@id="' + x +'"]/div/div/div[4]/span/span')

                        review_text5.append(text.get_attribute("innerHTML"))

                    except NoSuchElementException:

                        review_text5.append('')

                try:

                    driver.find_element("xpath","//ul[@class='a-pagination']/li[2]/a").click()

                    time.sleep(3)

                except : break

StaleElementReferenceException: Message: stale element reference: element is not attached to the page document
  (Session info: chrome=107.0.5304.108)
Stacktrace:
Backtrace:
	Ordinal0 [0x007D1ED3+2236115]
	Ordinal0 [0x007692F1+1807089]
	Ordinal0 [0x006766FD+812797]
	Ordinal0 [0x006792B4+823988]
	Ordinal0 [0x00679165+823653]
	Ordinal0 [0x00679AF5+826101]
	Ordinal0 [0x006D6452+1205330]
	Ordinal0 [0x006C1A8C+1120908]
	Ordinal0 [0x006D59E2+1202658]
	Ordinal0 [0x006C18A6+1120422]
	Ordinal0 [0x0069A73D+960317]
	Ordinal0 [0x0069B71F+964383]
	GetHandleVerifier [0x00A7E7E2+2743074]
	GetHandleVerifier [0x00A708D4+2685972]
	GetHandleVerifier [0x00862BAA+532202]
	GetHandleVerifier [0x00861990+527568]
	Ordinal0 [0x0077080C+1837068]
	Ordinal0 [0x00774CD8+1854680]
	Ordinal0 [0x00774DC5+1854917]
	Ordinal0 [0x0077ED64+1895780]
	BaseThreadInitThunk [0x761DFEF9+25]
	RtlGetAppContainerNamedObjectPath [0x77317BBE+286]
	RtlGetAppContainerNamedObjectPath [0x77317B8E+238]


In [271]:
title5

['Sometimes too worst experience with alexa she will not respond at a time',
 'Beware Not much useful Amazon going to stop Alexa',
 'Other Bluetooth speakers are better to handle than this device...',
 'Not as expected',
 "It mostly doesn't recognized",
 'Bad speaker quality n too low volume',
 'Don’t understand kannada words',
 'niche review padh lo',
 'Good product',
 "Doesn't supported YouTube.",
 '',
 '',
 'Voice and sound quality very poor',
 "YouTube can't be add in alexa skill. Hence most of Tamil songs not playing",
 'Not answer',
 "I don't like it",
 'Worst Purchase',
 'Not smart enough!',
 'Not improved with current technologies. Radios are not available.',
 'It is not connecting to mobile',
 'Fantastic sound quality but alexa is useless',
 'Sound',
 '',
 '',
 'Only bulb received.',
 'Disappointed with alexa device..',
 'Absolutely horrifying experience',
 'Not as expected. Flaws in design.',
 'Sound is not so good',
 'Poor speaker quality',
 'Not working....!!!',
 'Sound not

In [272]:
len(title5)

2958

In [273]:
review_text5

["I am giving to stars because the reason is that sometimes she will give too worst experience with us that she will not listen properly and if we say something to her sometimes she will not respond even the network is in good condition and she's having problem such things by Google that sometimes i ask questions so she will give a wrong answer i recommend you to update the mic setting for next user's because sometimes she will not catch voice even i am at nearer to her",
 'Not much useful as techie.. if u r a layman it might surprise u… but no daily news reader plug-in.. .news headline reading r dummy news..every day keep reading the same news...no horoscope.. no local language integration like telegu Tamil,, don’t find it useful except asking fewer things. Like joke whether.. switch on lights . friend working in Amazon tells they r going to stop Alexa no new development.. they made layoffs in Alexa team soon they r going to stop this product.. so beware ..',
 "It can't be operated wi

In [274]:
len(review_text5)

2958

In [275]:
ratings5

['2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '2.0',
 '',
 '',
 '2.0',
 '

In [276]:
len(ratings5)

2958

In [277]:
Title=title+title1+title2+title3+title4+title5

In [278]:
Review_Text=review_text+review_text1+review_text2+review_text3+review_text4+review_text5

In [279]:
Ratings=ratings+ratings1+ratings2+ratings3+ratings4+ratings5

In [280]:
print(len(Title),len(Review_Text),len(Ratings))

64259 64259 64259


In [281]:
import pandas as pd

In [282]:
df=pd.DataFrame({'Title':Title,'Review_Text':Review_Text,'Ratings':Ratings})
df

Title  \
0                                                          
1                                    Very bad experience   
2        मेरा product रिटर्न ले गया लेकिन रसीद नहीं दिया   
3                      Not up to the mark, disappointing   
4            Product quality, Sound quality, comfortness   
...                                                  ...   
64254                                               Good   
64255                                              Super   
64256  Very well built and compact in size but delive...   
64257                It's useful to smart home perfectly   
64258                                                      

                                             Review_Text Ratings  
0                                                                 
1      Anyone looking to buy this product please don'...     2.0  
2      मेरा पार्सल डिलिवरी बॉय २४/११/२२.को ले गया लेक...     2.0  
3      Everything is good but this product not up to ...     2.0  
4      Product quality - product quality is good in t...     2.0  
...                                                  ...     ...  
64254  Good product by Amazon,  very intelligent and ...     5.0  
64255                                              Super     5.0  
64256  I like it's magnificent sound quality and Alex...     5.0  
64257  It's very good. Nicely working without any pro...     5.0  
64258                                                             

[64259 rows x 3 columns]

In [283]:
df.head(15)

Title  \
0                                                       
1                                 Very bad experience   
2     मेरा product रिटर्न ले गया लेकिन रसीद नहीं दिया   
3                   Not up to the mark, disappointing   
4         Product quality, Sound quality, comfortness   
5                        Never buy wireless from boat   
6   Good for 1 year because the product I purchase...   
7                                         DOA product   
8                           Sound is not clear at all   
9                         Worst product worst company   
10              Silicon ear hooks loose and falls off   
11                                                      
12                                                      
13                                    Nahhh don't buy   
14                                       Disappointed   

                                          Review_Text Ratings  
0                                                              
1   Anyone looking to buy this product please don'...     2.0  
2   मेरा पार्सल डिलिवरी बॉय २४/११/२२.को ले गया लेक...     2.0  
3   Everything is good but this product not up to ...     2.0  
4   Product quality - product quality is good in t...     2.0  
5   I had brought this product about 10 days back....     2.0  
6   Actual battery percentage is not showing, it a...     2.0  
7   Can't tell if the Product is not working out o...     2.0  
8   I was so excited as got good feedback of the p...     2.0  
9   I have used this product , it's Been 9 months....     2.0  
10  Sound is good but can’t use the product as sil...     2.0  
11                                                             
12                                                             
13  Sound's good. The bass is good. Doesn't fit my...     2.0  
14  Used it for 1 year.  In the last one year have...     2.0

In [284]:
#saving this into system as a excel file
df.to_excel('AMAZON.xlsx')

##### I've got few errors as the website was taking too much time and since internet connectivity was not good at my place, so i tried to keep it simple by directly providing the url of the particular product in the url container to scrap out all the required details which was neceesary for the prediction of the target variable. 

##### I've extracted 64259 rows of data from amazon website but most of the rows are blank in this dataset i.e. it contains white spaces in the cells, so i'll be removing all the white spaces from this dataset while importing it into excel and further will load it into other jupyter notebook for predicting the attribute 'Ratings' on the basis of customers reviews.